### ISSUE TITLE
NoClassDefFoundError:  for DataBinderMapper even after enabling data binding

### ISSUE THREAD

**Issue by AkshayChordiya** [heinrichreimer/material-intro#220]
I'm facing this critical crash even after adding this

```
android {
    ....
    dataBinding {
        enabled = true
    }
}
```

Project Setup:
Using Android 3.0 Canary 5 with Kotlin and new Android architecture components

**Comment by heinrichreimer** [comment]
Is this related to #198? Which library and Android versions are you using? Is there a specific compiler error? If so, please post the full error log.

**Comment by AkshayChordiya** [comment]
Yeah it's related to #198 even after solving it. I am facing crash.

Details:
Library version: v1.6.2
Android version: 6.0.1 (Marshmallow)

Full error log:
```
FATAL EXCEPTION: main
                                                                               Process: com.bitfurther.notigo.teacher, PID: 23955
                                                                               java.lang.NoClassDefFoundError: Failed resolution of: Landroid/databinding/DataBinderMapper;
                                                                                   at android.databinding.DataBindingUtil.<clinit>(DataBindingUtil.java:31)
                                                                                   at com.heinrichreimersoftware.materialintro.app.IntroActivity.onCreate(IntroActivity.java:198)
                                                                                   at com.bitfurther.notigo.teacher.ui.activity.BoardingActivity.onCreate(BoardingActivity.kt:16)
                                                                                   at android.app.Activity.performCreate(Activity.java:6251)
                                                                                   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
                                                                                   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
                                                                                   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
                                                                                   at android.app.ActivityThread.-wrap11(ActivityThread.java)
                                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
                                                                                   at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                   at android.os.Looper.loop(Looper.java:148)
                                                                                   at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                                   at java.lang.reflect.Method.invoke(Native Method)
                                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                                Caused by: java.lang.ClassNotFoundException: Didn't find class "android.databinding.DataBinderMapper" on path: DexPathList[[zip file "/data/app/com.bitfurther.notigo.teacher-1/base.apk"],nativeLibraryDirectories=[/data/app/com.bitfurther.notigo.teacher-1/lib/arm, /vendor/lib, /system/lib]]
                                                                                   at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
                                                                                   at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
                                                                                   at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
                                                                                   at android.databinding.DataBindingUtil.<clinit>(DataBindingUtil.java:31) 
                                                                                   at com.heinrichreimersoftware.materialintro.app.IntroActivity.onCreate(IntroActivity.java:198) 
                                                                                   at com.bitfurther.notigo.teacher.ui.activity.BoardingActivity.onCreate(BoardingActivity.kt:16) 
                                                                                   at android.app.Activity.performCreate(Activity.java:6251) 
                                                                                   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107) 
                                                                                   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369) 
                                                                                   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) 
                                                                                   at android.app.ActivityThread.-wrap11(ActivityThread.java) 
                                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) 
                                                                                   at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                   at android.os.Looper.loop(Looper.java:148) 
                                                                                   at android.app.ActivityThread.main(ActivityThread.java:5417) 
                                                                                   at java.lang.reflect.Method.invoke(Native Method) 
                                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 
                                                                               	Suppressed: java.lang.ClassNotFoundException: android.databinding.DataBinderMapper
                                                                                   at java.lang.Class.classForName(Native Method)
                                                                                   at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
                                                                                   at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
                                                                                   at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
                                                                                   		... 16 more
                                                                                Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
```

**Comment by AkshayChordiya** [comment]
@heinrichreimer Need more data?

**Comment by wilsonqq** [comment]
Hi, @AkshayChordiya. Are you Using AIDE to compile and issue like this happen? It happens to me for several weeks, Android Studio is OK, but AIDE still occurred with this log.

**Comment by AkshayChordiya** [comment]
@wilsonqq I'm using Android Studio 3.0 Canary 5 and now 6. Still it's showing the same issue. I'll add some more code to debug the issue.

**Comment by daniel-stoneuk** [comment]
Try adding `android.enableAapt2=false` to your gradle.properties file

**Comment by kolbasan** [comment]
I am having the same issue, I am using similar setup, AS 3.0 Canary 6. 
Setting `android.enableAapt2=false` didn't fix the issue.

**Comment by AkshayChordiya** [comment]
@daniel-stoneuk Tried it, didn't work. Same crash issue.

**Comment by daniel-stoneuk** [comment]
@kolbasan @AkshayChordiya Ah damn, sorry. I presume you've had a Google to see whether it's solvable. I wonder whether this is an issue that should be resolved in the library, or if it's an external issue.

**Comment by AkshayChordiya** [comment]
Solved the issue by adding the following things to app `build.gradle`

1. Enabling Data Binding
```
android {
    ....
    dataBinding {
        enabled = true
    }
}
```


2. Adding dependency
```
dependencies {
    kapt 'com.android.databinding:compiler:2.3.3'
    ....
}
```
Using `kapt` for Kotlin.

**Comment by tasomaniac** [comment]
Same error happened to be but the `compiler` dependency is not available anymore for Gradle plugin version 3.2

**Comment by jwc123** [comment]
> Same error happened to be but the `compiler` dependency is not available anymore for Gradle plugin version 3.2

i had the wrong error when i upgrade the gradle plugin version to 3.2.Do you fix it?

**Comment by naimur20119655** [comment]
> Hi, @AkshayChordiya. Are you Using AIDE to compile and issue like this happen? It happens to me for several weeks, Android Studio is OK, but AIDE still occurred with this log.

AIDE can not generate DataBinding classes even after enabling databinding in build.gradle. Can you use databinding in AIDE?
